### PR TITLE
feat: add color option to file type svgs

### PR DESCRIPTION
**Pull request by panekj**

### CODE REVIEW

File: lapce-ui/src/svg.rs [hunk 1]
```diff
@@ -48,45 +48,82 @@ pub fn get_svg(name: &'static str) -> Option<Svg> {
...
 
-                for (exts, file_type) in TYPES {
-                    for ext in exts.iter() {
-                        if extension.eq_ignore_ascii_case(ext) {
-                            return Some(*file_type);
+                let (mut icon, mut color) = (None, None);
+
+                for (filenames, file_type, col) in FILES {
+                    for filename in filenames.iter() {
+                        if file_name.starts_with(filename) {
```
[the remark this answers is not in the document]
Comment (dzhou121) [reply lapce-ui/src/svg.rs:108]: Do you want to push this change?

**Comment by panekj** [comment]
Mind you that I put some colours for `.c`/`.h` & `.cpp`/`.hpp`, they might be not best looking or great as a default (:

**Comment by dzhou121** [comment]
> Mind you that I put some colours for `.c`/`.h` & `.cpp`/`.hpp`, they might be not best looking or great as a default (:

Actually can we remove the custom color for now. Hopefully we can get a designer in the future to do the icons and have different svgs for these.

On a sperate note, we also need different icon colors for light/dark theme. E.g. the Rust file icon doesn't look good on a dark theme.

**Comment by panekj** [comment]
> > Mind you that I put some colours for `.c`/`.h` & `.cpp`/`.hpp`, they might be not best looking or great as a default (:
> 
> Actually can we remove the custom color for now. Hopefully we can get a designer in the future to do the icons and have different svgs for these.

Sure do

> 
> On a sperate note, we also need different icon colors for light/dark theme. E.g. the Rust file icon doesn't look good on a dark theme.

Yes, I was thinking of replacing current icons with some much more wider set from someone who knows how to design stuff like https://github.com/file-icons/atom
I'll make thread on discord to gather what was already discussed and to talk about it more

**Pull request merged**